base_layer/mmr/src/balanced_binary_merkle_proof.rs

```diff
@@ -23,13 +23,14 @@
 use std::{collections::HashMap, convert::TryInto, marker::PhantomData};
 
 use digest::Digest;
+use serde::{Deserialize, Serialize};
 use tari_common::DomainDigest;
 use tari_utilities::ByteArray;
 use thiserror::Error;
 
 use crate::{common::hash_together, BalancedBinaryMerkleTree, Hash};
 
-#[derive(Debug)]
+#[derive(Deserialize, Serialize, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Default)]
```

Comment:
`PartialEq` and `Eq` are fine, but why `Ord` and `PartialOrd`? 

Also think we'll need Borsh encording

Reply:
To be consistent with the `MerkleProof` type. Glad to remove it, as well. Regarding Borsh encoding, makes sense, will add it